### PR TITLE
NRT

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/GraphQL.Authorization/AuthorizationValidationRule.cs [hunk 1]
```diff
@@ -135,7 +135,8 @@ public ValueTask<INodeVisitor> ValidateAsync(ValidationContext context)
                     // If some supplied field does not exist in the variable type then some other
                     // validation rule should check that but here we should just ignore that
                     // "unknown" field.
-                    if (context.Inputs.TryGetValue(variableRef.Name, out object input) &&
+                    if (context.Inputs != null &&
```
Comment (sungam3r) [inline src/GraphQL.Authorization/AuthorizationValidationRule.cs:138]: @Shane32 NRT helps find bug here - NRE before `context.Inputs` check, see `passes_with_claim_on_variable_type_without_inputs_but_fails_later_either`. I'll backport this check into `master`.

File: src/GraphQL.Authorization/AuthorizationValidationRule.cs [hunk 2]
```diff
@@ -172,7 +173,7 @@ public ValueTask<INodeVisitor> ValidateAsync(ValidationContext context)
             string errors = string.Join("\n", result.Errors);
 
             context.ReportError(new ValidationError(
-                context.Document.OriginalQuery,
+                context.Document.OriginalQuery!, // TODO: ???
```
Comment (sungam3r) [inline src/GraphQL.Authorization/AuthorizationValidationRule.cs:176]: @Shane32  I see that `Document.OriginalQuery` is `string?` now. How to deal with it in such cases?

Comment (sungam3r) [reply src/GraphQL.Authorization/AuthorizationValidationRule.cs:176]: `ValidationError` accepts `string`.

Comment (sungam3r) [reply src/GraphQL.Authorization/AuthorizationValidationRule.cs:176]: OK, I see :)
изображение

**Review by Shane32 — APPROVED** [review]

**Pull request merged**